sql/hive/pom.xml

```diff
@@ -199,6 +199,10 @@
       <artifactId>scalacheck_${scala.binary.version}</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.apache.maven.shared</groupId>
+      <artifactId>maven-invoker</artifactId>
```

Comment:
Seems too overkill to use Maven here, how about calling `javac` directly? https://github.com/apache/hive/blob/2d1405e7feed176aeed337581292b8438cf13326/common/src/java/org/apache/hive/common/util/HiveTestUtils.java#L94

Some additional backgrounds: there were some complains that Spark use two building tools, Maven and SBT, which introduces additional complex to align those behaviors, there is chance that one of the building tools might be dropped someday.

Reply:
It is necessary to build jar and not only compile Java class. Calling `javac` directly would require all the dependencies and without Maven it would be necessary to load them.

Using maven invoker here only impacts how test jar is built. It works both with Spark Maven and SBT builds and does not require usage of one over the other to build Spark.